### PR TITLE
feat(one-click-migration): add modal with farm options

### PR DESCRIPTION
**Pull request by FiboApe**

### CODE REVIEW

File: apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    >
+      {step === 'select-vaults' && (
+        <OneClickMigrationOptionsContent
+          onGoToStrategy={handleOnGoToStrategy}
+          onGoToDetails={handleOnGoToDetails}
+          farms={farms}
+        />
+      )}
+      {step === 'confirm-migration' && <div>Confirm migration</div>}
```
Comment (th0rOdinson) [inline apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx:62]: A Typography would suit better?

File: apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import usePushToHistory from '@hooks/usePushToHistory';
+import useTrackEvent from '@hooks/useTrackEvent';
+
+interface OneClickMigrationModalProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+const OneClickMigrationModal = ({ open, onClose }: OneClickMigrationModalProps) => {
+  const [step, setStep] = React.useState<'select-vaults' | 'confirm-migration'>('select-vaults');
```
Comment (th0rOdinson) [inline apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx:15]: If you were me, you would ask for an enum here 😄

File: apps/root/src/hooks/earn/useAvailableDepositTokens.tsx [hunk 3]
```diff
@@ -1,50 +1,81 @@
...
+          return acc;
+        }
 
-      const balances = tokenBalances.balances;
-      if (balances && Object.values(balances).some((balance) => balance > 0)) {
-        acc.push({
-          wallet: token.address,
-          token: { ...toToken({ ...token, price: tokenBalances.price }), farm: token.farm },
+        const balances = tokenBalances.balances;
+        const walletBalances = Object.entries(balances).filter(([, balance]) => balance > 0);
```
Comment (th0rOdinson) [inline apps/root/src/hooks/earn/useAvailableDepositTokens.tsx:53]: `> 0n` would be nice!

**Review by th0rOdinson — COMMENTED** [review]
I see a a few ts workarounds (i know this strategy stuff is type-hell), but if we could avoid some type assertions it may be beneficial for us in the future. Wouldn't spend much time on it though, just as a NTH

File: apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx (hunk 1 above)
Comment (FiboApe) [reply apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx:62]: This is just placeholder for the next ticket :)

File: apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx (hunk 2 above)
Comment (FiboApe) [reply apps/root/src/pages/earn/home/components/one-click-migration-modal/index.tsx:15]: Oooh good catch

**Review by th0rOdinson — APPROVED** [review]

**Pull request merged**